### PR TITLE
Message filter

### LINKED ISSUE #205: Advanced filtering of message feed
Needs #204.

The addition of severity data to the message feed will allow support for more advanced filtering of messages. We should support:

- filtering by severity value (depending on implementation of #204) possibly by a drop down of available values.
- free text filtering of message content.

### PR DESCRIPTION
**Pull request by Sahil590**
# Description
Adding advanced message feed filtering using htmx the same way we currently use it for the search. 
Fixes #205  (issue)

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

**Comment by Sahil590** [comment]
@cc-a I have addressed the severity filtering and the free text filtering is done by search, is there anything else that should be added?

File: main/templates/main/messages.html [hunk 1]
```diff
@@ -13,13 +13,31 @@ <h5 class="d-inline">Messages</h5>
...
+               placeholder="Search messages..."
+               hx-get="{% url 'main:messages' topic|default:"ALL" %}"
+               hx-trigger="input changed delay:500ms, every 1s"
+               hx-target="#message-list"
+               hx-include="#filter-form">
+        <!-- Severity Filter Dropdown -->
+        <select name="severity"
+                class="form-select mb-2"
+                hx-get="{% url 'main:messages' topic|default:"ALL" %}"
+                hx-trigger="change delay:500ms"
```
Comment (cc-a) [inline main/templates/main/messages.html:31]: don't think we need the delay here

**Review by cc-a — APPROVED** [review]
Looks good @Sahil590. One tiny suggested tweak to make it more responsive. We may need to consider a more sophisticated way to select multiple severities at once but lets get some user feedback first.

**Review by AdrianDAlessandro — APPROVED** [review]
I agree with Chris, being able to filter multiple severities (or filter _out_ individual ones) will probably be necessary, but this puts in an initial pass that we can get some feedback on.

File: main/templates/main/messages.html (hunk 1 above)
Comment (Sahil590) [reply main/templates/main/messages.html:31]: Yeah, I did try to play around with adding multiple but would need to add some custom JavaScript for that

Comment (Sahil590) [reply main/templates/main/messages.html]: I tried out Django filter but was having issues with the constant refreshing of the message pane. I could have found a workaround with custom JavaScript, but I thought this was cleaner.

File: main/templates/main/messages.html [hunk 2]
```diff
@@ -13,13 +13,30 @@ <h5 class="d-inline">Messages</h5>
...
+        <input class="form-control mb-2"
+               type="search"
+               name="search"
+               placeholder="Search messages..."
+               hx-get="{% url 'main:messages' topic|default:"ALL" %}"
+               hx-trigger="input changed delay:500ms, every 1s"
+               hx-target="#message-list"
+               hx-include="#filter-form">
+        <!-- Severity Filter Dropdown -->
+        <select name="severity"
```
Comment (dalonsoa) [inline main/templates/main/messages.html:28]: Far from being an expert in HTML, you should be able to select multiple options here by adding the `multiple` attribute. See https://www.w3schools.com/TAGS/att_select_multiple.asp Now idea what the choice look like in the backend, but you should be able to handle al the required logic there.

**Review by dalonsoa — COMMENTED** [review]
I think it should be pretty easy to handle multiple severities - see my comment. _Excluding_ certain severities is another story.

File: main/templates/main/messages.html (hunk 2 above)
Comment (dalonsoa) [reply main/templates/main/messages.html:28]: Ok, it won't work exactly as a dropdown, but can be tweaked a bit. See https://stackoverflow.com/a/65969350/3778792

Comment (cc-a) [reply main/templates/main/messages.html:28]: If we go this route then [Shoelace dropdowns](https://shoelace.style/components/dropdown) support having checkboxes. We'd need to figure out how it interacts with HTMX though.

**Comment by jamesturner246** [comment]
LGTM. For multiple severities, regex filtering could work: `^(debug|critical)$` etc, like in kafka topic: 

```python
regex = f"^({'|'.join(severities_to_show)})$"
DruncMessage.objects.filter(severity__regex=regex, ...
```

**Pull request merged**